### PR TITLE
Add remove_if_equal command to buildozer

### LINKED ISSUE #105: Buildozer doesn't have a way to remove an attribute if it has a specific value
I want to clear attributes that are set to a specific value. The use case is that an option was provided that had one default, and the default has switched. I now want to clear out all the settings where the new default is explicitly set.

### PR DESCRIPTION
**Pull request by dfreese**
I've left open the structure to removing more complex expressions, but trigger an error in all but the more basic cases.  I wasn't sure what sort of behavior this type of command should have in more complex cases.

I generate a comparison function up-front, which does a lot of virtual switching on types as there didn't seem to be a way to genericly compare expressions.  I've used getAttrValueExpr, however, this does make for a little weirdness in trying to handle the case of strings versus identifiers.  The test case with `"True"` versus `True` tries to demonstrate why identifiers are compared to both string and identifiers.

I assume this will seem incomplete in some respects, but I was putting this up for initial feedback on the approach as well as what would be expected in the test coverage.

Fixes #105

### CODE REVIEW

File: buildozer/buildozer_test.sh [hunk 1]
```diff
@@ -447,6 +447,42 @@ function test_remove_package_attribute() {
...
+
+function test_remove_if_equal_string_attr_string() {
+  in='go_library(
+    name = "edit",
+    toolchain = "something",
+)'
+  run "$in" 'remove_if_equal toolchain something' '//pkg:edit'
+  assert_equals 'go_library(name = "edit")'
+}
+
```
Comment (laurentlb) [inline buildozer/buildozer_test.sh:485]: You should add a negative test as well (where the value is not equal)

File: edit/buildozer.go [hunk 2]
```diff
@@ -417,6 +417,46 @@ func cmdRemove(opts *Options, env CmdEnvironment) (*build.File, error) {
 	return nil, nil
 }
 
+func cmdRemoveIfEqual(opts *Options, env CmdEnvironment) (*build.File, error) {
+	attr := env.Args[0]
+	args := env.Args[1:]
+	expr := getAttrValueExpr(attr, args, env)
+	cmp, err := func() (func(build.Expr) bool, error) {
```
Comment (laurentlb) [inline edit/buildozer.go:424]: Why is it in a function? Might be more readable to remove this function which is executed immediately.

File: edit/buildozer.go [hunk 3]
```diff
@@ -417,6 +417,46 @@ func cmdRemove(opts *Options, env CmdEnvironment) (*build.File, error) {
...
+				case *build.StringExpr:
+					return labels.Equal(val.Value, input.Name, env.Pkg)
+				case *build.Ident:
+					return val.Name == input.Name
+				default:
+					return false
+				}
+			}, nil
+		default:
+			return nil, fmt.Errorf("unsupported expression type %T", input)
```
Comment (laurentlb) [inline edit/buildozer.go:445]: Do we need to show an error here? I'd assume that the condition is false, so we can silently do nothing.

File: edit/buildozer.go [hunk 4]
```diff
@@ -417,6 +417,46 @@ func cmdRemove(opts *Options, env CmdEnvironment) (*build.File, error) {
 	return nil, nil
 }
 
+func cmdRemoveIfEqual(opts *Options, env CmdEnvironment) (*build.File, error) {
+	attr := env.Args[0]
+	args := env.Args[1:]
+	expr := getAttrValueExpr(attr, args, env)
```
Comment (laurentlb) [inline edit/buildozer.go:423]: getAttrValueExpr is using some heuristics to guess the expected type of an attribute. It is a bit brittle and I think we don't need this function here.

e.g. If you call `env.Rule.Attr` and get a string expression, you can just compare the content of the string with `attr`. No need to build an expr.

**Review by laurentlb — CHANGES_REQUESTED** [review]

File: buildozer/buildozer_test.sh (hunk 1 above)
Comment (dfreese) [reply buildozer/buildozer_test.sh:485]: done

File: edit/buildozer.go (hunk 4 above)
Comment (dfreese) [reply edit/buildozer.go:423]: done

File: edit/buildozer.go (hunk 2 above)
Comment (dfreese) [reply edit/buildozer.go:424]: I tend to find immediately invoked functions, and a single assignment more clear.  I tend to write more C++ where that allows the variable to be const, but that isn't an advantage here.  Thanks for pointing that out; removed.

File: edit/buildozer.go (hunk 3 above)
Comment (dfreese) [reply edit/buildozer.go:445]: done

**Review by dfreese — COMMENTED** [review]
Thanks for the review, and sorry for the delay in my response.  Personal matters came up.

**Review by vladmos — CHANGES_REQUESTED** [review]
Looks good, but please also document the new command in buildozer/README.md

**Comment by dfreese** [comment]
> Looks good, but please also document the new command in buildozer/README.md

Thanks.  Done.

**Review by vladmos — APPROVED** [review]
Thanks!

**Review by laurentlb — APPROVED** [review]

**Pull request merged**